pydropsonde/helper/quality.py

```diff
@@ -76,6 +76,28 @@ def get_is_floater(
         )
         return surface_ds.time[0].values
 
+    def profile_extend(
+        self,
+        ds,
+        alt_dim="gpsalt",
+        extend_ratio=0.1,
+    ):
+        variables = self.qc_vars
+        for variable in variables:
+            drop_alt = ds.attrs["aircraft_msl_altitude_(m)"]
+            no_na = ds.dropna(dim="time", subset=[variable])[alt_dim].values
+            if no_na.size > 0:
+                max_alt = np.nanmax(no_na)
+            else:
+                max_alt = np.nan
+
+            max_diff = drop_alt - max_alt
+            if max_diff > drop_alt * float(extend_ratio):
+                self.qc_flags[f"{variable}_profile_extend"] = False
+            else:
+                self.qc_flags[f"{variable}_profile_extend"] = True
```

Comment:
You could remove the if statement
```suggestion
            self.qc_flags[f"{variable}_profile_extend"] = max_diff < drop_alt * float(extend_ratio)
```